--- Makefile.old	2013-03-12 02:37:04.073104742 +0100
+++ Makefile.in	2013-03-12 02:37:17.642935096 +0100
@@ -645,7 +645,8 @@
 # builders
 builders_residfp_builder_libsidplayfp_residfp_ladir = $(includedir)/sidplayfp/builders
 builders_residfp_builder_libsidplayfp_residfp_la_HEADERS = \
-builders/residfp-builder/residfp.h
+builders/residfp-builder/residfp.h \
+builders/residfp-builder/residfp-emu.h
 
 builders_residfp_builder_libsidplayfp_residfp_la_SOURCES = \
 builders/residfp-builder/residfp.cpp \
@@ -658,7 +659,8 @@
 builders_residfp_builder_libsidplayfp_residfp_la_LDFLAGS = -version-info $(LIBRESIDFPVERSION)
 builders_resid_builder_libsidplayfp_resid_ladir = $(includedir)/sidplayfp/builders
 builders_resid_builder_libsidplayfp_resid_la_HEADERS = \
-builders/resid-builder/resid.h
+builders/resid-builder/resid.h \
+builders/resid-builder/resid-emu.h
 
 builders_resid_builder_libsidplayfp_resid_la_SOURCES = \
 builders/resid-builder/resid.cpp \
@@ -671,7 +673,8 @@
 builders_resid_builder_libsidplayfp_resid_la_LDFLAGS = -version-info $(LIBRESIDVERSION)
 @HARDSID_TRUE@builders_hardsid_builder_libsidplayfp_hardsid_ladir = $(includedir)/sidplayfp/builders
 @HARDSID_TRUE@builders_hardsid_builder_libsidplayfp_hardsid_la_HEADERS = \
-@HARDSID_TRUE@builders/hardsid-builder/hardsid.h
+@HARDSID_TRUE@builders/hardsid-builder/hardsid.h \
+@HARDSID_TRUE@builders/hardsid-builder/hardsid-emu.h
 
 @HARDSID_TRUE@@MINGW32_FALSE@hardsid_src = builders/hardsid-builder/hardsid-unix.cpp
 @HARDSID_TRUE@@MINGW32_TRUE@hardsid_src = builders/hardsid-builder/hardsid-win.cpp
