--- src/qvortaro.h-old	2008-02-22 07:50:23.000000000 +0100
+++ src/qvortaro.h	2009-08-17 16:33:41.000000000 +0200
@@ -114,4 +114,5 @@
 
 };
 
+Q_DECLARE_METATYPE(QAction *)
 #endif // QVORTARO_H
--- src/qvortaro.cpp-old	2008-02-21 22:38:08.000000000 +0100
+++ src/qvortaro.cpp	2009-08-17 16:33:46.000000000 +0200
@@ -498,7 +498,7 @@
   tmp.insert("size", size());
   tmp.insert("savePosSize", true);
 
-  tmp.insert("zeigeLanguage2", m_aAnsichtZeigeLanguage2);
+  tmp.insert("zeigeLanguage2", QVariant::fromValue(m_aAnsichtZeigeLanguage2));
 
   if (m_agLexika1->checkedAction())
     tmp.insert("lexika1", m_agLexika1->checkedAction()->data());
