--- configure.old	2006-03-05 22:01:49.000000000 +0100
+++ configure	2009-08-17 18:27:13.000000000 +0200
@@ -1737,7 +1737,7 @@
 
 echo "$as_me:$LINENO: checking for udev version" >&5
 echo $ECHO_N "checking for udev version... $ECHO_C" >&6
-udev_ver=`udevinfo -V | awk '{ print $3 }'`
+udev_ver=`udevadm --version`
 if test -n "$udev_ver"; then
   echo "$as_me:$LINENO: result: $udev_ver" >&5
 echo "${ECHO_T}$udev_ver" >&6
