diff -urpN wt-3.3.3.orig/src/Wt/Dbo/SqlQueryParse.C wt-3.3.3/src/Wt/Dbo/SqlQueryParse.C
--- wt-3.3.3.orig/src/Wt/Dbo/SqlQueryParse.C	2015-02-04 12:57:24.640624711 +0100
+++ wt-3.3.3/src/Wt/Dbo/SqlQueryParse.C	2015-02-04 13:04:35.222851079 +0100
@@ -20,7 +20,7 @@
 #include <boost/spirit/include/phoenix_core.hpp>
 #include <boost/spirit/include/phoenix_operator.hpp>
 #include <boost/spirit/include/phoenix_object.hpp>
-#include <boost/spirit/home/phoenix/statement/throw.hpp>
+#include <boost/phoenix/statement/throw.hpp>
 #include <boost/bind.hpp>
 #include <iostream>
 
diff -urpN wt-3.3.3.orig/src/Wt/Json/Parser.C wt-3.3.3/src/Wt/Json/Parser.C
--- wt-3.3.3.orig/src/Wt/Json/Parser.C	2015-02-04 12:57:24.652624885 +0100
+++ wt-3.3.3/src/Wt/Json/Parser.C	2015-02-04 12:59:26.850391907 +0100
@@ -24,7 +24,7 @@
 #include <boost/spirit/include/phoenix_core.hpp>
 #include <boost/spirit/include/phoenix_operator.hpp>
 #include <boost/spirit/include/phoenix_object.hpp>
-#include <boost/spirit/home/phoenix/statement/throw.hpp>
+#include <boost/phoenix/statement/throw.hpp>
 #include <boost/bind.hpp>
 
 #endif // JSON_PARSER
diff -urpN wt-3.3.3.orig/src/Wt/Render/CssParser.C wt-3.3.3/src/Wt/Render/CssParser.C
--- wt-3.3.3.orig/src/Wt/Render/CssParser.C	2015-02-04 12:57:24.648624826 +0100
+++ wt-3.3.3/src/Wt/Render/CssParser.C	2015-02-04 13:03:05.197549282 +0100
@@ -34,7 +34,7 @@ using namespace Wt::Render;
 #include <boost/fusion/include/adapt_struct.hpp>
 #include <boost/variant/recursive_variant.hpp>
 #include <boost/spirit/home/support/context.hpp>
-#include <boost/spirit/home/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/classic_file_iterator.hpp>
 
 #include <map>
@@ -157,8 +157,7 @@ struct fs_error_tag {};
 template< typename Iterator >
 struct ErrorReporting
 {
-  template< typename, typename, typename, typename, typename > // Phoenix v2
-  struct result { typedef void type;};
+  typedef void result_type;
 
   ErrorReporting(CssGrammer< Iterator >* grammer)
     : grammer_(grammer)
