Index: yoshimi/src/UI/BankUI.fl
===================================================================
--- yoshimi.orig/src/UI/BankUI.fl	2011-07-10 11:30:47.000000000 -0400
+++ yoshimi/src/UI/BankUI.fl	2011-07-10 11:45:57.000000000 -0400
@@ -51,7 +51,10 @@
 class BankProcess_ {open
 } {
   Function {process(void)} {open return_type {virtual void}
-  } {}
+  } {
+    comment {This body intentionally left blank.} {in_source not_in_header
+    }
+  }
   decl {Bank *bank;} {public
   }
 }
Index: yoshimi/src/UI/OscilGenUI.fl
===================================================================
--- yoshimi.orig/src/UI/OscilGenUI.fl	2011-07-10 11:30:47.000000000 -0400
+++ yoshimi/src/UI/OscilGenUI.fl	2011-07-10 11:42:21.000000000 -0400
@@ -166,7 +166,11 @@
 
 class PSlider {open : {public Fl_Slider}
 } {
-  Function {PSlider(int x,int y, int w, int h, const char *label=0):Fl_Slider(x,y,w,h,label)} {} {}
+  Function {PSlider(int x,int y, int w, int h, const char *label=0):Fl_Slider(x,y,w,h,label)} {
+  } {
+    comment {This body intentionally left blank.} {in_source not_in_header
+    }
+  }
   Function {handle(int event)} {return_type int
   } {
     code {int X=x(),Y=y(),W=w(),H=h();
Index: yoshimi/src/UI/PartUI.fl
===================================================================
--- yoshimi.orig/src/UI/PartUI.fl	2011-07-10 11:30:47.000000000 -0400
+++ yoshimi/src/UI/PartUI.fl	2011-07-10 11:46:36.000000000 -0400
@@ -89,7 +89,10 @@
 class PartUI_ {open
 } {
   Function {showparameters(int kititem,int engine)} {open return_type virtual
-  } {}
+  } {
+    comment {This body intentionally left blank.} {in_source not_in_header
+    }
+  }
 } 
 
 class PartKitItem {: {public Fl_Group}
Index: yoshimi/src/UI/PresetsUI.fl
===================================================================
--- yoshimi.orig/src/UI/PresetsUI.fl	2011-07-10 11:30:47.000000000 -0400
+++ yoshimi/src/UI/PresetsUI.fl	2011-07-10 11:50:44.000000000 -0400
@@ -44,8 +44,10 @@
 
 class PresetsUI_ {} {
   Function {refresh()} {return_type {virtual void}
-  } {}
-  Function {~PresetsUI_()} {} {}
+  } {
+    comment {This body intentionally left blank.} {in_source not_in_header
+    }
+  }
 } 
 
 class PresetsUI {} {

 
