--- scribus/plugins/tools/2geomtools/lib2geom/CMakeLists.txt.old	2009-04-23 02:29:46.920745916 -0700
+++ scribus/plugins/tools/2geomtools/lib2geom/CMakeLists.txt	2009-04-23 02:31:52.794615295 -0700
@@ -15,6 +15,7 @@
 #CB SET(CMAKE_CXX_FLAGS "-Wall")
 #CB SET(CMAKE_MAKE_PROGRAM "${CMAKE_MAKE_PROGRAM}")
 # We need to pass -fPIC only to lib2geom on amd64 - avoid it elsewhere see: http://www.gentoo.org/proj/en/base/amd64/howtos/index.xml?part=1&chap=3
+SET(CMAKE_CXX_FLAGS "-fPIC")
 
 OPTION(2GEOM_BUILD_SHARED
   "Build lib2geom and libtoy as shared libraries."

